js/bucket.js

```diff
@@ -22,19 +54,19 @@ import { assert, noGroupsLabel } from './util.js';
  *  the value.
  */
 export function dateBuckets(jsonData, fieldName, start, end, interval) {
-  let startDt = moment(start);
-  let endDt = moment(end);
+  let startDt = DateTime.fromISO(start);
+  let endDt = DateTime.fromISO(end);
 
-  assert(startDt.isValid(), 'Start date is invalid');
-  assert(endDt.isValid(), 'End date is invalid');
+  assert(startDt.isValid, 'Start date is invalid');
+  assert(endDt.isValid, 'End date is invalid');
 
   // Groups the data by interval start and summarizes the counts.
   let countsByInterval = jsonData.reduce((counts, item) => {
-    let dt = moment(item[fieldName]);
+    let dt = DateTime.fromISO(item[fieldName]);
 
-    if (dt.isSameOrAfter(startDt) && dt.isSameOrBefore(endDt)) {
-      let intervalStart = moment(dt.startOf(interval));
-      let key = intervalStart.format("YYYY-MM-DD");
```

Comment:
I was puzzled by this comparison at first, since JavaScript doesn't have operator overloading. This works by [overriding the `valueOf` method](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/valueOf) on the `DateTime` object, which JS uses when coercing values. I thought this was a good technique to know about.